app/src/ui/changes/commit-message-avatar.tsx

```diff
@@ -0,0 +1,226 @@
+ */
+export class CommitMessageAvatar extends React.Component<
+  ICommitMessageAvatarProps,
+  ICommitMessageAvatarState
+> {
+  public constructor(props: ICommitMessageAvatarProps) {
+    super(props)
+
+    const account = this.props.repositoryAccount
+
+    this.state = {
+      isPopoverOpen: false,
+      accountEmail: account !== null ? lookupPreferredEmail(account) : '',
+    }
+  }
+
+  private shouldShowMisattributedCommitWarning() {
+    if (
+      this.props.email === undefined ||
+      this.props.repositoryAccount === null ||
+      this.props.repository === null ||
+      this.props.repository.ignoreWrongUserEmail
+    ) {
+      return false
+    }
+
+    const emails = this.props.repositoryAccount.emails.map(e => e.email)
+    return emails.includes(this.props.email) === false
+  }
+
+  public render() {
+    return (
+      <div className="commit-message-avatar-component">
+        <div onClick={this.onAvatarClick}>
+          {this.shouldShowMisattributedCommitWarning() &&
+            this.renderMisattributedCommitWarning()}
+          <Avatar user={this.props.user} title={this.props.title} />
+        </div>
+        {this.state.isPopoverOpen && this.renderPopover()}
+      </div>
+    )
+  }
+
+  private renderMisattributedCommitWarning() {
+    return (
+      <div className="misattributed-commit-warning-badge">
+        <Octicon symbol={OcticonSymbol.alert} />
+      </div>
+    )
+  }
+
+  private openPopover = () => {
+    this.setState(prevState => {
+      if (prevState.isPopoverOpen === false) {
+        return { isPopoverOpen: true }
+      }
+      return null
+    })
+  }
+
+  private closePopover = () => {
+    this.setState(prevState => {
+      if (prevState.isPopoverOpen) {
+        return { isPopoverOpen: false }
+      }
+      return null
+    })
+  }
+
+  private onAvatarClick = (event: React.FormEvent<HTMLDivElement>) => {
+    if (this.shouldShowMisattributedCommitWarning() === false) {
+      return
+    }
+
+    event.preventDefault()
+    if (this.state.isPopoverOpen) {
+      this.closePopover()
+    } else {
+      this.openPopover()
+    }
+  }
+
+  private renderPopover() {
+    const account = this.props.repositoryAccount
+    if (account === null) {
+      return
+    }
+
+    const allEmails = account.emails.map(e => e.email)
+
+    const accountTypeSuffix =
+      account.endpoint === getDotComAPIEndpoint() ? '' : ' Enterprise'
+
+    return (
+      <Popover
+        caretPosition={PopoverCaretPosition.LeftTop}
+        onClickOutside={this.closePopover}
+      >
+        <h3>This commit will be misattributed</h3>
+        <Row>
+          <div>
+            The email in your{' '}
+            <span className="highlighted-text">git config</span> (
+            {this.props.email}) doesn't match your{' '}
+            <span className="highlighted-text">
+              GitHub{accountTypeSuffix} account
+            </span>
```

Comment:
I think having these chunks of text with the same color as a link might be misleading. Specially since there is an actual link in the same popover. I picked this text style from the mocks, but maybe we should do something else… 🤔 

What do you think, @ampinsk?

![image](https://user-images.githubusercontent.com/1083228/108055702-9e45c800-7050-11eb-88e4-8525f77b1738.png)

Reply:
:D - I was going to comment this.. I tried to click them...